swagger/coordinator.yml

```diff
@@ -0,0 +1,214 @@
+          description: If the coordinator can't yet provide the content, the response body will be empty.
+  /participant/sum2:
+    post:
+      tags:
+        - Coordinator
+      description: PET Step 12 (send "update" message)
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Sum2Message"
+      responses:
+        200:
+          description: OK
+        400:
+          description: PET protocol error
+  /global_model:
+    get:
+      tags:
+        - Aggregator
+      description: Download the global model
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: string
+                format: byte
+        404:
+          description: model not found
+components:
+  schemas:
+    SumMessage:
+      description: Sum message
+      type: object
+      required:
+        - message
+      properties:
+        message:
+          type: string
+          format: byte
+    UpdateMessage:
+      description: Update message
+      type: object
+      required:
+        - message
+      properties:
+        message:
+          type: string
+          format: byte
+    Sum2Message:
+      description: Sum2 message
+      type: object
+      required:
+        - message
+      properties:
+        message:
+          type: string
+          format: byte
+    PublishRoundMessage:
+      type: object
+      properties:
+        coord_pk:
+          description: The public key of the coordinator.
+          type: string
+          format: byte
+          minLength: 32
+          maxLength: 32
+        sum:
+          description: The expected fraction of participants selected for computing the unmasking sum.
+          type: number
+          format: float
+          minimum: 0
+          maximum: 1
+        update:
+          description: The expected fraction of participants selected for submitting an updated local model for aggregation.
+          type: number
+          format: float
+          minimum: 0
+          maximum: 1
+        seed:
+          description: Is a random block of bits used for the selection of participants.
+          type: string
+          format: byte
+          minLength: 32
+          maxLength: 32
+        global_model_url:
+          description: The URL that points to the global model. Participants can use this URL to fetch the global model. On the first round, the value of the URL will be null.
+          type: string
+          format: url
+          nullable: true
+    PublishSumParticipantsMessage:
+      type: object
+      additionalProperties:
+        type: string
+        format: byte
+    PublishEncryptedSeedsMessage:
+      type: object
+      additionalProperties:
+        type: array
+        items:
+          type: string
+          format: byte
+          minLength: 32
+          maxLength: 32
```

Comment:
this should be of min/max length `80`.

this is because the encryption of the seed produces a header of 48 bytes (32 bytes for the public part of an ephemeral key and 16 bytes for a MAC). the ciphertext of the seed itself stays 32 bytes, but without the header it is undecryptable even for the intended receiver.

Reply:
Fixed